README.md

```diff
@@ -116,3 +116,67 @@ func testNonEquatableFailure() {
 
 - Support for working with `Scheduler`s to avoid relying on timeouts
 
+## AutomaticallyEquatable
+
+When you have a complex type or type graph that you would like to compare for unit testing purposes, you can use this protocol by extending your type to conform to it.
+This will drastically reduce the volume of code required to make unit test equality assertions on custom types.
+It also negates the need to rely on custom `Equatable` implementations.
+(Custom `Equatable` implementations come with the risk that future changes to the type may invalidate the equatable implementation without warning.)
+
+**Important Disclosures**
+
+This is an imperfect and assuming implementation of `Equatable`. It should not be used without understanding the following concepts.
+
+The implementation:
+
+- Cannot anticipate the consequences of observing a calculated property. (ie, code that changes the state of data when a property is observed).
+- Cannot respect custom `Equatable` implementations of the values being compared or any of the subsequent members. It will use its own comparison logic instead.
+- Skips over members that cannot be reasonably compared, such as closures. These are assumed to be equal.
```

Comment:
I wonder if you could deal with closures such that if the _addresses_ to which two closures point, then they are equal, otherwise, they are not.

Reply:
I think that should be possible. It May be possible to use the same approach to prevent recursion as well (lookup table of a history memory addresses within the type graph of the compared values)

Unfortunately, I don't have time to go down those rabbit holes. The current version gets the job done with some limitations. Definitely worth some investigation down the road!